src/lua/rust_interop.rs

```diff
@@ -0,0 +1,174 @@
+//! Rust code which is called from lua in the init file
+#![deny(unused_code)]
+
+use std::sync::Arc;
+use std::ops::Deref;
+
+use hlua::{self, Lua, LuaTable};
+use hlua::any::AnyLuaValue;
+use hlua::any::AnyLuaValue::*;
+use rustc_serialize::json::Json;
+
+use registry::{self, RegistryField, RegistryError, AccessFlags};
+use commands;
+use keys::{self, KeyPress, KeyEvent};
+use convert::json::{json_to_lua, lua_to_json};
+use convert::serialize::ToTable;
+
+type OkayResult = Result<(), &'static str>;
+type ValueResult = Result<AnyLuaValue, &'static str>;
+
+/// We've `include!`d the code which initializes from the Lua side.
+const LUA_INIT_CODE: &'static str = include_str!("../../lib/lua/lua_init.lua");
```

Comment:
Though it would probably be optimized out, should this really be a `const`? It'll only be used in one place in this entire module and it's not even exported.